src/deprecates.jl

```diff
@@ -48,3 +48,15 @@ findat(a::AbstractArray, b::AbstractArray) = findat!(Array{Int}(size(b)), a, b)
 @deprecate df_residual(obj::StatisticalModel) dof_residual(obj)
 
 @deprecate_binding WeightVec Weights
+
+using Base.Random: RangeGenerator
+@deprecate_binding RandIntSampler RangeGenerator
+@deprecate RangeGenerator(K::Int) RangeGenerator(1:K)
```

Comment:
I'm not so concerned about the absence of a deprecation warning: it's unlikely that a package would use the type without any of the associated functions, which do print warnings. But however the type would be named, how would ensure at the same type that the old code continues to work (both code using the type and code using the functions)? AFAICT that requires type piracy.